### DISCUSSION TITLE
I deleted my cache directory and now images are blank mostly

### DISCUSSION THREAD

**Discussion by Semisol** [mastodon/mastodon#16806] [Q&A]
I deleted my cache directory without knowing there is a command to clear the cache, is there a way I can fix images being blank?

**Comment by ClearlyClaire** [comment]
There is no built-in way to handle this, but https://github.com/mastodon/mastodon/pull/16641 should help

**Reply by Semisol** [reply]
Any updates on this?

**Reply by Semisol** [reply]
I did a mistake and deleted remote media from the database too, I don't know what to do now...

**Reply by ClearlyClaire** [reply]
Run `RAILS_ENV=production bundle exec rails c` to open a Rails console, and in it, paste:

```ruby
MediaAttachment.cached.where.not(remote_url: '').each do |attachment|
  attachment.file.destroy
  attachment.thumbnail.destroy
  attachment.save
end
```

**Reply by travnewmatic** [reply]
curious if theres a good solution for this now?

i havent deleted my media (yet), but i am preparing to rebuild my homelab and i need to temporarily store my media remotely.

it seems like a good portion of my media is in the `cache/` directory.  i'm assuming that content is not data that is unique to my instance i.e. content that i've uploaded to my server.

as mentioned in https://github.com/mastodon/mastodon/pull/16641 it would be nice to have a way to only backup the stuff that is really unique to my instance, and re-download whatever is missing from federated instances when i bring my instance back online.

i dont have much experience using `RAILS_ENV=production bundle exec rails c` so im a little wary.  is [this](https://github.com/mastodon/mastodon/discussions/16806#discussioncomment-1659609) still the recommended approach?

i know it comes with the territory of selfhosting, but i must have backed-up my mastodon media a dozen times and it takes foreeeever because `media/` is a gajillion tiny files in a gajillion directories.  and when i look at the filenames like `cache/accounts/headers/106/388/089/076/020/349/original/562432e6c17e6569.png` i kiiiiiinda feel like i'm wasting my time and should get on with the migration already.

any help would be appreciated and thanks so much for all the hard work on this project!~  its great!

**Reply by YJSoft** [reply]
@travnewmatic Use `tootctl media remove` command. You can safely remove remote cache media without breaking anything.

**Reply by travnewmatic** [reply]
> @travnewmatic Use `tootctl media remove` command. You can safely remove remote cache media without breaking anything.

alright, i'm impatient enough to give it a try, but just to confirm, is that `media_dir/cache`?

what about `media_dir/accounts`?  i feel like i have a lot of crap in there too

and thank you very very much for the response!

**Reply by travnewmatic** [reply]
also, i do periodically run `tootctl media remove` and `tootctl media remove-orphans`

because i have so much stuff in `mediadir/cache`, did i screw up my media?  is that leftover from a long time ago and mastodon doesnt know about it so i isnt getting deleted?

**Reply by ClearlyClaire** [reply]
`tootctl media remove` should remove a good chunk of what is in `media_dir/cache`, as well as older media attachments in `media_dir/media_attachments`; however:
- it does not remove *all* media attachments but “only” those older than 7 days by default
- it only deals with media attachments, not custom emoji, account avatars or profile headers

Unfortunately, there is currently no tool that completely removes everything from the cache and alters the database to mark them as needing a redownload.

**Reply by travnewmatic** [reply]
> `tootctl media remove` should remove a good chunk of what is in `media_dir/cache`, as well as older media attachments in `media_dir/media_attachments`; however:
> 
> * it does not remove _all_ media attachments but “only” those older than 7 days by default
> * it only deals with media attachments, not custom emoji, account avatars or profile headers
> 
> Unfortunately, there is currently no tool that completely removes everything from the cache and alters the database to mark them as needing a redownload.

i gotcha, thanks for the response @ClearlyClaire :)

**Reply by falavu** [reply]
Hello,
where i have to type in this Command? 
I did it in the Terminal  but it says "tootctl: command not found". 
What i am doing wrong?

**Reply by StrangeMinds1960** [reply]
su - mastodon -s /bin/bash
cd live/bin
RAILS_ENV=production ./tootctl media remove
